src/libostree/ostree-repo-commit.c

```diff
@@ -486,12 +486,7 @@ create_regular_tmpfile_linkable_with_content (OstreeRepo *self,
           if (bytes_read < 0)
             return FALSE;
           else if (bytes_read == 0)
-            {
-              if (G_UNLIKELY (remaining > 0))
-                return glnx_throw (error, "Unexpected EOF with %" G_GUINT64_FORMAT "/%" G_GUINT64_FORMAT " bytes remaining", remaining, length);
-              else
-                break;
-            }
+            return glnx_throw (error, "Unexpected EOF with %" G_GUINT64_FORMAT "/%" G_GUINT64_FORMAT " bytes remaining", remaining, length);
```

Comment:
Actually, no, the code pushed in #1037 is wrong. It still redundantly checks `(remaining > 0)` when that condition is guaranteed by the `while` condition.